REFACTORING-ISSUE-228.md

```
@@ -0,0 +1,251 @@
+- Separate CLI Agent detection from persistence service
+
+### Long-term Architecture
+- Consider implementing Command Pattern for all user actions
+- Evaluate dependency injection container for better testability
+- Establish architectural decision records (ADRs) for future changes
+
+---
+
+## References
+
+- **Issue #228**: https://github.com/s-hiraoku/vscode-sidebar-terminal/issues/228
```

markdownlint-cli2 (0.18.1)

Bare URL used

(MD034, no-bare-urls)